src/binary-writer.cc

```diff
@@ -97,10 +97,97 @@ struct RelocSection {
   std::vector<Reloc> relocations;
 };
 
-struct Symbol {
-  Index symbol_index;
-  SymbolType type;
-  Index element_index;
+class Symbol {
+ public:
+  struct Function {
+    static const SymbolType type = SymbolType::Function;
+    Index index;
+  };
+  struct Data {
+    static const SymbolType type = SymbolType::Data;
+    Index index;
+    Offset offset;
+    Address size;
+  };
+  struct Global {
+    static const SymbolType type = SymbolType::Global;
+    Index index;
+  };
+  struct Section {
+    static const SymbolType type = SymbolType::Section;
+    Index section;
```

Comment:
Why not call this `index` like the others?   Or the other way around and don't call any of them `index`?

Reply:
I was using the names from https://github.com/WebAssembly/tool-conventions/blob/master/Linking.md, where this field is called "section".